service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1344,14 +1326,16 @@ private Order incrementCounter(Order order) {
     private PaymentRequestDto formPayment(Long orderId, int sumToPay) {
         Order order = orderRepository.findById(orderId)
             .orElseThrow(() -> new OrderNotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+        int lastNumber = order.getPayment().size() - 1;
         PaymentRequestDto paymentRequestDto = PaymentRequestDto.builder()
             .merchantId(Integer.parseInt(merchantId))
             .orderId(
-                orderId + order.getCounterOrderPaymentId().toString() + order.getPayment().get(0).getId())
+                orderId + "_" + order.getCounterOrderPaymentId().toString() + "_"
+                    + order.getPayment().get(lastNumber).getId())
```

Comment:
make sure order.getPayment().get(lastNumber).getId() is NPE safe and no out of bound possible